frontend/src/pages/flags.page.tsx

```diff
@@ -0,0 +1,163 @@
+      return apiFetch(`/flags/${existingFlag.id}/`, {
+        method: "PATCH",
+        body: JSON.stringify({
+          everyone: enable,
+        }),
+      });
+    }
+
+    return await apiFetch("/flags/", {
+      method: "POST",
+      body: JSON.stringify({
+        name: flagName,
+        everyone: enable,
+      }),
+    });
+  }
+
+  const onSubmit: FormEventHandler = async (event) => {
+    event.preventDefault();
+
+    if (!["enable", "disable"].includes(actionInput)) {
+      return;
+    }
+
+    const enable = actionInput === "enable";
+    const response = await createOrUpdate(flagInput, enable);
+    flagData.mutate();
+    if (response.ok) {
+      toast(
+        <>
+          Flag <output>{flagInput}</output> {enable ? "enabled" : "disabled"}{" "}
+          successfully
+        </>,
+        { type: "success" }
+      );
+      // Don't make it too easy to update multiple flags in a row
+      // (to avoid making mistakes):
+      setActionInput("");
+      setFlagInput("");
+    } else {
+      toast(
+        <>
+          Something went wrong {enable ? "enabling" : "disabling"} flag{" "}
+          <output>{flagInput}</output>
+        </>,
+        { type: "error" }
+      );
+    }
+    return;
+  };
+
+  return (
+    <Layout runtimeData={runtimeData.data}>
+      <main className={styles.wrapper}>
+        <table className={styles["flag-list"]}>
+          <thead>
+            <tr>
+              <th>Active?</th>
+              <th>Flag</th>
+            </tr>
+          </thead>
+          <tbody>
+            {flags.map((flag) => (
+              <tr
+                key={flag.name}
+                className={
+                  flag.everyone ? styles["is-active"] : styles["is-inactive"]
+                }
+              >
+                <td>
+                  {flag.everyone ? (
+                    <CheckIcon alt="Active" />
+                  ) : (
+                    <BlockIcon alt="Inactive" />
+                  )}
+                </td>
+                <td>{flag.name}</td>
+              </tr>
+            ))}
+          </tbody>
+        </table>
+        <form onSubmit={onSubmit} className={styles["flag-form"]}>
+          <p>
+            To avoid accidentally enabling/disabling the wrong flags, you have
+            to type out explicitly what action to take.
+          </p>
+          <div className={styles.field}>
+            <label htmlFor="action">Which flag do you want to modify?</label>
+            <input
+              type="text"
+              id="action"
+              autoComplete="off"
+              required={true}
+              placeholder={`e.g. \`${flags[0]?.name}\``}
+              value={flagInput}
+              onChange={(e) => setFlagInput(e.target.value)}
+            />
+          </div>
+          <div className={styles.field}>
+            <label htmlFor="action">
+              Do you want to <output>enable</output> or <output>disable</output>{" "}
```

Comment:
I'd like a third option. Setting `everyone` to `true` or `false` turns it on or off globally, like a switch. `null` means there is no global option, and you can set or clear the flag at the user or group level or as a percent.

Reply:
Hmm, so I started looking at this, but both the code as well as the UI got pretty complex pretty quickly, and then I realised that maybe this wouldn't be that useful anyway, since you can't currently set non-global flags.

What I did do, however, is to warn people when they try to modify a non-global flag, to avoid them discarding individual settings. If those have been set (i.e. by SRE), then from then on the flag can only be modified by SRE. bb542af1dea318d03fe98413f361fb9b969b88e9

Reply:
That makes sense to me. We can add more subtle option when we iterate.